src/strategy.rs

```diff
@@ -275,29 +278,225 @@ pub fn hostname<S: BuildHasher>(parameters: Option<HashMap<String, String, S>>)
+                    let value = getter(context)
+                        .map(|v| {
+                            as_vec.iter().any(|x| {
+                                if case_insensitive {
+                                    v.to_lowercase().starts_with(x)
+                                } else {
+                                    v.starts_with(x)
+                                }
+                            })
+                        })
+                        .unwrap_or(false);
+                    _handle_inversion(&value, &inverted)
+                })
+            }
+        }
+        ConstraintExpression::StrEndsWith {
+            values,
+            case_insensitive,
+        } => {
+            if values.is_empty() {
+                Box::new(move |_| _handle_inversion(&false, &inverted))
+            } else {
+                let mut as_vec: Vec<String> = values.to_vec();
+                let case_insensitive = case_insensitive.unwrap_or(false);
+                if case_insensitive {
+                    as_vec = as_vec.iter_mut().map(|x| x.to_lowercase()).collect();
+                };
+                Box::new(move |context: &Context| {
+                    let value = getter(context)
+                        .map(|v| {
+                            as_vec.iter().any(|x| {
+                                if case_insensitive {
+                                    v.to_lowercase().ends_with(x)
+                                } else {
+                                    v.ends_with(x)
+                                }
+                            })
+                        })
+                        .unwrap_or(false);
+                    _handle_inversion(&value, &inverted)
+                })
+            }
+        }
+        ConstraintExpression::StrContains {
+            values,
+            case_insensitive,
+        } => {
+            if values.is_empty() {
+                Box::new(|_| false)
+            } else {
+                let mut as_vec: Vec<String> = values.to_vec();
+                let case_insensitive = case_insensitive.unwrap_or(false);
+                if case_insensitive {
+                    as_vec = as_vec.iter_mut().map(|x| x.to_lowercase()).collect();
+                };
+                Box::new(move |context: &Context| {
+                    let value = getter(context)
+                        .map(|v| {
+                            as_vec.iter().any(|x| {
+                                if case_insensitive {
+                                    v.to_lowercase().contains(x)
+                                } else {
+                                    v.contains(x)
+                                }
+                            })
+                        })
+                        .unwrap_or(false);
+                    _handle_inversion(&value, &inverted)
+                })
+            }
+        }
+        ConstraintExpression::NumEq { value } => match value.parse::<f64>() {
```

Comment:
and this family of contraints should also be extractable into composable helpers